app/frontend/src/cljs/teet/asset/cost_items_controller.cljs

```diff
@@ -470,3 +478,24 @@
     (callback)
     (t/fx app
           common-controller/refresh-fx)))
+
+(defn filtered-cost-group-totals
+  "Return ATL item that is being filtered by and cost group totals that match
+  the filter.
+
+  Adds :ui/group to each item that contains a vector of type hierarchy."
+  [app atl cost-group-totals]
+  (let [kw (some-> app (get-in [:query :filter]) cljs.reader/read-string)
```

Comment:
Had to check that this can't be used to read arbitrary Clojurescript code. :D It's essentially an edn reader.

Reply:
I had the same concern and checked, perhaps should comment that so future readers don't get raised blood pressure